packages/astro/src/@types/astro.ts

```diff
@@ -1096,12 +1087,111 @@ export interface AstroAdapter {
 
 type Body = string;
 
-export interface APIContext {
+// Shared types between `Astro` global and API context object
+interface AstroSharedContext<Props extends Record<string, any> = Record<string, any>> {
+	/**
+	 * The address (usually IP address) of the user. Used with SSR only.
+	 */
+	clientAddress: string;
+	/**
+	 * Utility for getting and setting cookies values.
```

Comment:
Nit but “cookies values” would probably need a possessive apostrophe — “cookies’ values” — which can be tricky on plurals like this, so might be clearer just to use “of” here:

```suggestion
	 * Utility for getting and setting the values of cookies.
```